apps/server/src/tasks/tasks.service.ts

```diff
@@ -140,32 +147,132 @@ export class TasksService {
+        skip: getTasksInput.offset * getTasksInput.limit,
+        order: {
+          chunkInfo: {
+            start: 'ASC' as const,
+          },
+        },
+        take: getTasksInput.limit,
+      });
+    } else {
+      const floatTasks = await Task.find({
+        relations: {
+          // Include chunks in case of float tasks
+          chunks: true,
+          chunkInfo: {
+            repeat: true,
+          },
+        },
+        where: {
+          // Get those values from filterOptions that are in task entity
+          ...Object.keys(getTasksInput?.filterOptions || {}).filter(
+            (key) => key in Task
+          ),
+          ...(getTasksInput?.filterOptions?.category && {
+            category: In(getTasksInput.filterOptions.category),
+          }),
+          ...(getTasksInput?.filterOptions?.priority && {
+            priority: In(getTasksInput.filterOptions.priority),
+          }),
+          ...(typeof getTasksInput?.filterOptions?.repeat === 'boolean' && {
+            chunkInfo: {
+              repeat: getTasksInput?.filterOptions?.repeat
+                ? Not(IsNull())
+                : IsNull(),
+            },
+          }),
+          isFloat: true,
+          user: { id: user.id },
+        },
+        skip: getTasksInput.offset * getTasksInput.limit,
+        order: {
+          chunkInfo: {
+            start: 'ASC' as const,
+          },
+        },
+        take: getTasksInput.limit,
+      });
+
+      const constTasks = await Task.find({
+        relations: {
+          chunkInfo: {
+            repeat: true,
+          },
+        },
+        where: {
+          // Get those values from filterOptions that are in task entity
+          ...Object.keys(getTasksInput?.filterOptions || {}).filter(
+            (key) => key in Task
+          ),
+          ...(getTasksInput?.filterOptions?.category && {
+            category: In(getTasksInput.filterOptions.category),
+          }),
+          ...(getTasksInput?.filterOptions?.priority && {
+            priority: In(getTasksInput.filterOptions.priority),
+          }),
+          ...(typeof getTasksInput?.filterOptions?.repeat === 'boolean' && {
+            chunkInfo: {
+              repeat: getTasksInput?.filterOptions?.repeat
+                ? Not(IsNull())
+                : IsNull(),
+            },
+          }),
+          isFloat: false,
+          user: { id: user.id },
+        },
+        skip: getTasksInput.offset * getTasksInput.limit,
+        order: {
+          chunkInfo: {
+            start: 'ASC' as const,
+          },
+        },
+        take: getTasksInput.limit,
+      });
+      return [...floatTasks, ...constTasks];
+    }
   }
 
   async getTask(user: User, id: string) {
     return await Task.findOne({
       relations: ['chunks', 'chunkInfo', 'chunkInfo.repeat'],
       where: {
-        user: user,
-        id: id,
+        user: { id: user.id },
+
+        // Convert id to integer
+        id: parseInt(id),
```

Comment:
can we get rid of those comments?

Reply:
Probably yes, but don't think it's urgent.